RESTserver/src/main/java/com/softserve/if072/restservice/controller/StoreController.java

```diff
@@ -123,16 +123,50 @@ public void deleteStore(@PathVariable int id, HttpServletResponse response) thro
    @GetMapping("/{id}/products")
    @ResponseBody
    @ResponseStatus(value = HttpStatus.OK)
-   public List<Product> getAllProducts(@PathVariable int id, HttpServletResponse response) throws IOException {
+   public List<Product> getAllProductsFromStore(@PathVariable int id, HttpServletResponse response) {
         try {
             List<Product> products = storeService.getProductsByStoreId(id);
             LOGGER.info("All Products were found");
             return products;
-        } catch (DataSourceException e) {
-            response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
             LOGGER.error("Products were not found", e);
             return null;
         }
    }
 
+    @GetMapping("/{storeId}/products/{productId}")
+    @ResponseBody
+    @ResponseStatus(value = HttpStatus.OK)
+    public Product getProductFromStore(@PathVariable int storeId, int productId, HttpServletResponse response) {
+        try {
+            Product product = storeService.getProductFromStoreById(storeId, productId);
+            LOGGER.info("All Products were found");
+            return product;
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error((String.format("Product %d from Store %d not found", productId, storeId)), e);
+            return null;
+        }
+    }
+
+    @DeleteMapping("/{storeId}/products/{productId}")
+    @ResponseStatus(value = HttpStatus.NO_CONTENT)
+    public void deleteProductFromStore(@PathVariable int storeId, int productId, HttpServletResponse response) {
+        try {
+            storeService.deleteProductFromStoreById (storeId, productId);
+            LOGGER.info(String.format("Product %d from Store %d was deleted", productId, storeId));
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error((String.format("Product %d from Store %d not found", productId, storeId)), e);
```

Comment:
As your exception contains the same message, no need to create it again. just call e.getMessage()